app/components/GeoLocationInput/GeoLocationModal/GeoLocationMapInput/GeoLocationMap/index.tsx

```diff
@@ -0,0 +1,299 @@
+import React, { useMemo, useState, useCallback } from 'react';
+import { _cs, isDefined, isNotDefined } from '@togglecorp/fujs';
+import Map, {
+    MapBounds,
+    MapContainer,
+    MapSource,
+    MapLayer,
+    MapTooltip,
+    MapState,
+} from '@togglecorp/re-map';
+import {
+    Spinner,
+    SegmentInput,
+    Container,
+    TextOutput,
+    List,
+} from '@the-deep/deep-ui';
+import {
+    MapboxGeoJSONFeature,
+    AnySourceData,
+    MapboxOptions,
+    Layer,
+    LngLat,
+    PopupOptions,
+    LngLatLike,
+} from 'mapbox-gl';
+import {
+    GeoAreaBounds,
+} from '#types';
+import { useRequest } from '#base/utils/restRequest';
+import _ts from '#ts';
+
+import { AdminLevel } from '../index';
+import styles from './styles.css';
+
+const scaleFactor = 1;
+const sourceOptions: AnySourceData = {
+    type: 'geojson',
+    // FIXME: we need to think about this
+    promoteId: 'pk', // NOTE mapbox requires each feature property to have a unique identifier. Right now the server adds pk to each property so we are using it.
+};
+
+const fillLayerOptions: Omit<Layer, 'id'> = {
+    type: 'fill',
+    paint: {
+        'fill-opacity': ['case',
+            ['boolean', ['feature-state', 'hovered'], false],
+            0.7,
+            0.5,
+        ],
+        'fill-color': ['case',
+            ['boolean', ['feature-state', 'selected'], false],
+            '#1a3ed0',
+            '#a9bedc',
```

Comment:
Do we need to create constants for maps in deep? @frozenhelium @samshara @AdityaKhatri

Reply:
@AdityaKhatri any thoughts?